cupy/core/_routines_statistics.pyx

```diff
@@ -315,9 +318,26 @@ cpdef ndarray _median(
 
     keep_ndim = a.ndim
 
+    out_shape = None
+    if sequence.PySequence_Check(axis):
+        # cupy.sort and cupy.partition only support integer axis, so move
+        # all reduced dimensions to the end and reshape them into a single
+        # reduction axis.
+        reduce_axis, out_axis = _reduction._get_axis(axis, keep_ndim)
+        out_shape = _reduction._get_out_shape(a.shape, reduce_axis, out_axis,
+                                              keepdims)
+        a = a.transpose(out_axis + reduce_axis)
+        sort_shape = tuple([a.shape[n] for n in range(len(out_axis))]) + (-1,)
+        a = a.reshape(sort_shape)
+        if not a.flags.c_contiguous:
+            a = cupy.ascontiguousarray(a)
+        axis = -1
+
     if axis is None:
         sz = a.size
     else:
+        if axis < -keep_ndim or axis >= keep_ndim:
+            raise numpy.AxisError('Axis overrun')
```

Comment:
nit: how about raising a cupy exception instead of numpy.
If AxisError is not available, ValueError should do it.

Reply:
There are several other `.pyx` files in `cupy/core` (e.g. `core/_routines_sorting.pyx`) that do raise `numpy.AxisError`, so I was following that example. (`cupy.AxisError` does not seem to exist)

Did you mean to try something like:
```Python
try:
    raise numpy.AxisError('Axis overrun')
except AttributeError:
    raise ValueError('Axis overrun')
```
or just to raise `ValueError` in the first place?

Reply:
There used to be an `_AxisError` in `cupy/core/_errors.py`, but #3096 removed it. I don't get why.

Reply:
>  I don't get why.

I think it was just deemed unnecessary now that CuPy no longer supports older NumPy predating `numpy.AxisError`

Reply:
Sorry perhaps I'm being stupid 😅 Doesn't NumPy still have `numpy.AxisError` in newer versions?

Reply:
Yes. It was introduced in 1.13. That is why I think it seems a bit odd to add a try/except like above at this point.

Should CuPy define a `cupy.AxisError` in its API? (perhaps as a subclass of numpy.AxisError). Adding that would best be addressed in a separate PR, though.

Reply:
If other files raise it then I am ok with it.
Sorry for not verifying.

We should re-add the exception and homogenize everything in a different PR.

Thanks!